onnx/common/path.cc

```diff
@@ -0,0 +1,22 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+// Copyright (c) ONNX Project Contributors.
+// Licensed under the MIT license.
+
+#include "onnx/common/path.h"
+#include "onnx/string_utils.h"
+
+namespace ONNX_NAMESPACE {
+
+const std::string path_join(const std::string& origin, const std::string& append) {
+    std::string new_path = origin;
+    if (new_path.length() >= k_preferred_path_separator.length() && 
+        new_path.substr(new_path.length () - k_preferred_path_separator.length(), k_preferred_path_separator.length()) != k_preferred_path_separator) {
+        new_path += k_preferred_path_separator;
```

Comment:
this check can be simplified something like
if(origin.find_last_of(k_prefferred_path_separator) == origin.length() - k_preferred_path_separator.length()) {
return origin + k_preferred_path_separator + append ;
}
return origin + append

this way you can avoid creating multiple intermediate strings

Reply:
Good idea. Updated.